packages/immutable-rootfs/30cos-immutable-rootfs/cos-mount-layout.sh

```diff
@@ -109,10 +109,13 @@ function readCOSLayoutConfig {
     # An empty RW_PATHS is a valid value, default rw_paths are only 
     # applied when RW_PATHS is unset.
     if [ -n "${RW_PATHS+x}" ]; then
-        rw_paths=(${RW_PATHS})
+        rw_paths=(${RW_PATHS[@]})
     fi
     if [ -n "${PERSISTENT_STATE_PATHS}" ]; then
-        state_paths=(${PERSISTENT_STATE_PATHS})
+        state_paths=(${PERSISTENT_STATE_PATHS[@]})
+        for path in "${state_paths[@]}"; do
+            [ ! -d "${path}" ] && echo "Path ${path} from PERSISTENT_STATE_PATHS does not exist!" && exit 0
```

Comment:
I think we should rather check that on the rootfs, in this way it's going to check the ones in the initramfs? cc @davidcassany

Reply:
1. I agree with assigning VARIABLES to array and I have dropped this change

2. I have added additional check to functions `mountOverlay`and `mountState`. Now it will check first if source mount path exists, if not then will return from function and continue